internal/conditions/conditions.go

```diff
@@ -21,7 +21,7 @@ const (
 	ReasonGatewayNotReady             = "GatewayNotReady"
 	ReasonGatewayReady                = "GatewayReady"
 	ReasonMaxPipelinesExceeded        = "MaxPipelinesExceeded"
-	ReasonReferencedSecretMissing     = "ReferencedSecretMissing"
+	ReasonReferencedSecretInvalid     = "ReferencedSecretInvalid"
```

Comment:
IMHO, the new name does not reflect the reality? Is the secret invalid or secret reference invalid? The new name suggests the former.

Reply:
I agree, but do you have a better suggestion?
Maybe `SecretReferenceInvalid`?

Reply:
It should be a union of `SecretReferenceInvalid` and `SecretKeyMissing`. Maybe `UnableToAccessSecret` or something (please, discuss with @a-thaler)

Reply:
I agree with @a-thaler, especially that this is a corner case that will be prevented by the CRD validation in the future. So it's more like the error message that customers currently in this situation see. Given this, changing the name of one of our "most popular" reasons names may not be the best idea.